### ISSUE TITLE
Merging different shape inputs & predicting labels from multiple output layers

### ISSUE THREAD

**Issue by spate141** [keras-team/keras#10225]
- [X] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [X] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

![image](https://user-images.githubusercontent.com/10580847/40188189-04dca954-59bf-11e8-8943-8088e6aeff2f.png)

I'm trying to implement a model with functional API where output of `Dense Output_0` layer is feed back as input to separate/multiple sub-model like layers to predict labels from each of those different 3 Dense/Softmax layers. 

Issue is, I don't know how to combine `two variable shape inputs` in a way it helps connected sub-model based parts to predict labels corresponding to output from `Dense Output_0`.

> For Example:

Lets assume, `Dense Output_0 is [0.9, 0.9, 0.1]`, so here `Hidden 1` & `Hidden 2` sub-model parts will be triggered and model predicts labels from `Dense Output_1` & `Dense Output_2`. I can understand that `Dense Output_3` will also predict labels but here I want to make those predictions to be really irrelevant for this particular one since it got input `0.1`.

I'm sorry if this is something which is already discussed. In that case, I would really appreciate if anyone can point me to relevant documents/issues.

**Comment by spate141** [comment]
Closing the issue as the above situation can be handle by some engineering efforts on Keras `Merge`  layers and proper use of Functional API.

:)
